core/src/main/java/com/linecorp/armeria/client/InvalidResponseHeadersException.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.HttpHeaders;
+
+/**
+ * An {@link InvalidResponseException} raised when a client received a response with invalid headers.
+ */
+public class InvalidResponseHeadersException extends InvalidResponseException {
```

Comment:
How about naming as `InvalidResponseStatusException` because this is raised only when the status is not `OK`?

Reply:
Perhaps this exception could be used somewhere else? I deliberately made it broadly applicable.